messages/java/src/main/java/io/cucumber/messages/Messages.java

```diff
@@ -0,0 +1,4417 @@
+
+        public UndefinedParameterType(
+            String expression,
+            String name
+        ) {
+            this.expression = expression;
+            this.name = name;
+            validate();
+        }
+
+
+        public String getExpression() {
+            return expression;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        private void validate() {
+            java.util.Objects.requireNonNull(expression, "UndefinedParameterType.expression cannot be null");
+            java.util.Objects.requireNonNull(name, "UndefinedParameterType.name cannot be null");
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            UndefinedParameterType that = (UndefinedParameterType) o;
+            return 
+                expression.equals(that.expression) &&             
+                name.equals(that.name);            
+        }
+
+        @Override
+        public int hashCode() {
+            return java.util.Objects.hash(
+                expression,
+                name
+            );
+        }
+
+        @Override
+        public String toString() {
+            return "UndefinedParameterType{" +
+                "expression=" + expression +
+                ", name=" + name +
+                '}';
+        }
+    }
+
+    public enum AttachmentContentEncoding {
+        IDENTITY("IDENTITY"),
+        BASE64("BASE64");
+    
+        private final String value;
+
+        AttachmentContentEncoding(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return this.value;
+        }
+
+        public String value() {
+            return this.value;
+        }
+
+        public static AttachmentContentEncoding fromValue(String value) {
+            if ("IDENTITY".equals(value)) return IDENTITY;
+            if ("BASE64".equals(value)) return BASE64;
+            throw new IllegalArgumentException(value);
+        }
+    }
+
+    public enum SourceMediaType {
+        TEXT_X_CUCUMBER_GHERKIN_PLAIN("text/x.cucumber.gherkin+plain"),
+        TEXT_X_CUCUMBER_GHERKIN_MARKDOWN("text/x.cucumber.gherkin+markdown");
+    
+        private final String value;
+
+        SourceMediaType(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return this.value;
+        }
+
+        public String value() {
+            return this.value;
+        }
+
+        public static SourceMediaType fromValue(String value) {
+            if ("TEXT_X_CUCUMBER_GHERKIN_PLAIN".equals(value)) return TEXT_X_CUCUMBER_GHERKIN_PLAIN;
+            if ("TEXT_X_CUCUMBER_GHERKIN_MARKDOWN".equals(value)) return TEXT_X_CUCUMBER_GHERKIN_MARKDOWN;
+            throw new IllegalArgumentException(value);
```

Comment:
This implementation is equivalent to the `Enu.valueOf`. Did you mean to compare against `.value`? 

https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#valueOf(java.lang.Class,%20java.lang.String)

Reply:
Yeah just noticed - fixed in 288ba0c06750542ce061bb1480c3d6ffef04a289 on #1879